server/auth/auth.go

```diff
@@ -1178,6 +1180,103 @@ func (auth *AuthManager) handleMatchStatus(conn comms.Link, msg *msgjson.Message
 	return nil
 }
 
+// marketOrders is an index of order IDs associated with a particular market.
+type marketOrders struct {
+	base     uint32
+	quote    uint32
+	orderIDs map[order.OrderID]bool
+}
+
+// add adds a match ID to the marketMatches.
+func (mo *marketOrders) add(oid order.OrderID) bool {
+	_, found := mo.orderIDs[oid]
+	mo.orderIDs[oid] = true
+	return !found
+}
+
+// idList generates a []order.OrderID from the currently indexed order IDs.
+func (mo *marketOrders) idList() []order.OrderID {
+	ids := make([]order.OrderID, 0, len(mo.orderIDs))
+	for oid := range mo.orderIDs {
+		ids = append(ids, oid)
+	}
+	return ids
+}
+
+// handleOrderStatus handles requests to the 'order_status' route.
+func (auth *AuthManager) handleOrderStatus(conn comms.Link, msg *msgjson.Message) *msgjson.Error {
+	client := auth.conn(conn)
+	if client == nil {
+		return msgjson.NewError(msgjson.UnauthorizedConnection,
+			"cannot use route 'order_status' on an unauthorized connection")
+	}
+
+	var orderReqs []msgjson.OrderStatusRequest
+	err := json.Unmarshal(msg.Payload, &orderReqs)
+	if err != nil {
+		return msgjson.NewError(msgjson.RPCParseError, "error parsing order_status: %v", err)
+	}
```

Comment:
Just occurring to me, but shouldn't clients sign the order_status and match_status request payloads? Would be necessary if we want to limit requesting details for orders and matches to those pertaining to a user's account alone.